homeassistant/util/read_only_dict.py

```diff
@@ -14,6 +14,10 @@ def _readonly(*args: Any, **kwargs: Any) -> Any:
 class ReadOnlyDict(dict[_KT, _VT]):
     """Read only version of dict that is compatible with dict types."""
 
+    def __hash__(self) -> int:  # type: ignore[override]
+        """Return the hash of the dict."""
+        return id(self)
```

Comment:
This won't work since its created every time we fire

We could do `hash(frozenset(my_dict.items()))` but that would only work if its not nested